### ISSUE TITLE
Customization overloads HA and crashes it

### ISSUE THREAD

**Issue by eavyon** [Blackymas/NSPanel_HA_Blueprint#1348]
I have setup a new nspanel with latest update in dev 4.1.2 with this great blueprint. 4.1.1 didnt want to update the TFT, see other issue. Im running ESP-IDF. Trouble started when I added the customization "set display as a light". 

The part "id: !extend set_brightness" makes a loop setting the brightness over and over again as can be seen in debug log. Im not able to debug right now but after removing only that part of the code the debug log is quiet again and HA stays up.
If someone able to reproduce it because I dont understand why it keeps calling this script and setting brightness as fast as it can.

Log repeats this part:
[15:25:11][D][script.set_brightness:1700]: brightness: 69%
[15:25:11][D][script.set_brightness(custom):138]: brightness: 69
[15:25:11][D][light:036]: 'nspanel Display' Setting:
[15:25:11][D][light:051]:   Brightness: 69%

**Comment by edwardtfn** [comment]
I will look at that.

**Comment by edwardtfn** [comment]
I believe that is fixed now.
Could you please update your code with the one available now on the Customization docs and try again?

**Comment by eavyon** [comment]
After many tries I have to conclude your updated code has fixed this particular problem. Except only after adding your code and compiling again the Nextion screen isnt detected anymore. It stays now forever on boot screen with TFT 4.1.1. I tried almost everything. Running 4.1.1 and dev 4.1.2. Esp-IDF and arduino framework. This isnt probably related but it happened directly after only that simple code update. Will look into it tomorrow and open new issue probably. :(

**Comment by edwardtfn** [comment]
The code on dev is currently broken and I have exactly the same issue you reported.
Beta and main are working fine.

**Comment by edwardtfn** [comment]
>  It stays now forever on boot screen with TFT 4.1.1.

OK, TFT version is 4.1.1, but what about esphome version shown in that screen?

**Comment by eavyon** [comment]
That is 4.2dev because I used your pr#5825 earlier. And I did try a TFT upload because the screen got recognised again. But also that and reverting to 4.1.2 did not recover it. 4.1.2 doesnt recognise the screen.
Added log with what happened.
[231128_nspanel.txt](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/13482285/231128_nspanel.txt)

**Comment by eavyon** [comment]
I would say close this issue because it seems the current problem is diagnosed here https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/1350 Running beta on arduino now and screen is working again.
